tests/online_ddl/conf/dm-task.yaml

```diff
@@ -73,8 +73,12 @@ filters:
   filter-rule-index:
     schema-pattern: "*"
     table-pattern: "*"
-    events: ["create index","drop index"]
-    sql-pattern: ["ALTER\\s.*TABLE[\\s\\S]*ADD\\s+KEY\\s+NAME", "ALTER\\s.*TABLE[\\s\\S]*DROP\\s+KEY"]
+    events: ["create index", "drop index"]
+    sql-pattern:
+      [
+        "alter table(.*?)add (key|index)(.*?)name",
```

Comment:
should not change original test case if no strong reason. User executed `ADD KEY` so he will write an `ADD KEY` filter

Reply:
user raw sql is  `alter table t2 add key name (name);`
this sql will be rewrite to 
```
ALTER TABLE `online_ddl`.`_t1_new` add key name (name)
```
in pt, and this sql cannot be filtered by the binlog filter

The reason we were able to pass the test before is that this sql was ignored for `ignore, err := s.skipQuery(nil, nil, sql)` (table name:`_t1_new` not match with block-allow-list)

---
but in new logic, we don't `skipQuery` twice, so we need update this binlog filter rule

Reply:
as I said in https://github.com/pingcap/dm/pull/1824#discussion_r663740810 if you can't find a way to properly keep the old behaviour and no strong reasons, don't delete that skip

we should change the code instead of requiring users to change their usages.